pkg/plugins/backendplugin/plugin_settings.go

```diff
@@ -0,0 +1,45 @@
+package backendplugin
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type pluginSettings map[string]string
+
+func (ps pluginSettings) ToEnv(prefix string, hostEnv []string) []string {
+	env := []string{}
+	for k, v := range ps {
+		env = append(env, fmt.Sprintf("%s_%s=%s", prefix, strings.ToUpper(k), v))
+	}
+
+	env = append(env, hostEnv...)
+
+	return env
+}
+
+func extractPluginSettings(cfg *setting.Cfg) map[string]pluginSettings {
+	pluginSettings := map[string]pluginSettings{}
+	for _, section := range cfg.Raw.Sections() {
```

Comment:
Instead of looping through all the settings, I wonder if `extractPluginSettings` could get as an argument only the plugin section.
Probably `setting` package could export specifically the plugin section.